actions/submit-signing-request/task.ts

```diff
@@ -231,32 +232,35 @@ export class Task {
             }
         });
 
-        const targetFilePath = path.join(process.env.GITHUB_WORKSPACE as string, this.signedArtifactDestinationPath)
+        const targetDirectory = this.resolveOrCreateDirectory(this.outputArtifactDirectory);
 
-        // make sure that the target directory exists
-        const targetDir = path.dirname(targetFilePath);
-        if (!fs.existsSync(targetDir)) {
-            core.info(`Directory ${targetDir} does not exist, and will be created`);
-            fs.mkdirSync(targetDir, { recursive: true });
-        }
+        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetDirectory}`);
 
-        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetFilePath}`);
-        const writer = fs.createWriteStream(targetFilePath)
+        // save the signed artifact to temp ZIP file
+        const tmpZipFile = path.join(targetDirectory, '__signpath_signed_artifact_tmp.zip');
+        const writer = fs.createWriteStream(tmpZipFile)
```

Comment:
Not necessary, but since you're consistent otherwise: colon (;) missing at the end ;-)

Reply:
Fixed